resource-managers/kubernetes/integration-tests/src/test/scala/org/apache/spark/deploy/kubernetes/integrationtest/backend/GCE/GCETestBackend.scala

```diff
@@ -22,18 +22,19 @@ import org.apache.spark.deploy.kubernetes.config.resolveK8sMaster
 import org.apache.spark.deploy.kubernetes.integrationtest.backend.IntegrationTestBackend
 import org.apache.spark.deploy.kubernetes.integrationtest.constants.GCE_TEST_BACKEND
 
-class GCETestBackend extends IntegrationTestBackend {
-  val master = System.getProperty("spark.kubernetes.test.master")
-  var k8ConfBuilder = new ConfigBuilder()
-    .withApiVersion("v1")
-    .withMasterUrl(resolveK8sMaster(master))
-  val k8ClientConfig = k8ConfBuilder.build
-  val defaultClient = new DefaultKubernetesClient(k8ClientConfig)
+private[spark] class GCETestBackend(val master: String) extends IntegrationTestBackend {
+  private var defaultClient: DefaultKubernetesClient = _
```

Comment:
In the GCE case it's probably fine to initialize in the constructor - create a KubernetesClient should be fairly lightweight. What we wanted to avoid in the Minikube case was to have side effects in the constructor.